rakelib/pod.rake

```diff
@@ -0,0 +1,21 @@
+require 'json'
+
+def current_pod_version
+  JSON.parse(File.read(PODSPEC_FILE))['version']
+end
+
+namespace :pod do
+  # rake pod:lint
+  desc "Lint the podspec"
+  task :lint do
+    header "Linting podspec"
+    sh("pod", "lib", "lint", PODSPEC_FILE)
```

Comment:
```suggestion
    sh("bundle", "exec", "pod", "lib", "lint", PODSPEC_FILE)
```

Reply:
I don't actually think this is necessary — and will actually not work as you think if used that way.
I base this on the fact that I previously forgot to add `gem 'cocoapods'` to the `Gemfile` and got an error about "Can't find pod in the bundle" or something. So I think the shell command run with `sh` is executed in the context of Bundler already.

Reply:
You sure about that? Maybe you don't have CocoaPods installed system-wide?

Reply:
Well that's also confirmed by the existence of `Bundler.with_clean_env do … end` which exists in order to run commands _outside_ of the bundler env when needed… as opposed to running inside the bundler env by default. See also [this SO answer](https://stackoverflow.com/questions/24284347/execute-system-command-with-rake-outside-bundle-scope/26410873).

---

Proof by testing: given this task I temporarily added to the `Rakefile`:

```rake
task :test do
  sh("pod", "--version")
  sh("bundle", "exec", "pod", "--version")
end
```

<details><summary>without CocoaPods in the <tt>Gemfile</tt></summary>

```sh
$ cat Gemfile
# frozen_string_literal: true

source "https://rubygems.org"

gem "octokit"
gem "rake"
```
```sh
$ bundle install
Using rake 13.0.1
Using public_suffix 4.0.5
Using addressable 2.7.0
Using bundler 2.1.4
Using multipart-post 2.1.1
Using faraday 1.0.1
Using sawyer 0.8.2
Using octokit 4.18.0
Bundle complete! 2 Gemfile dependencies, 8 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

```sh
$ be rake test
pod --version
Traceback (most recent call last):
	2: from /Users/olivier/.gem/ruby/2.6.0/bin/pod:23:in `<main>'
	1: from /Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/rubygems_integration.rb:402:in `block in replace_bin_path'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/rubygems_integration.rb:374:in `block in replace_bin_path': can't find executable pod for gem cocoapods. cocoapods is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
rake aborted!
Command failed with status (1): [pod --version...]
/Users/olivier/Documents/Dev/GitHub/Stencil/Rakefile:13:in `block in <top (required)>'
/Users/olivier/.gem/ruby/2.6.0/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in `load'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in `kernel_load'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:28:in `run'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli.rb:476:in `exec'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli.rb:30:in `dispatch'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/base.rb:476:in `start'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/cli.rb:24:in `start'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/exe/bundle:46:in `block in <top (required)>'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/lib/bundler/friendly_errors.rb:123:in `with_friendly_errors'
/Users/olivier/.gem/ruby/2.6.0/gems/bundler-2.1.4/exe/bundle:34:in `<top (required)>'
/usr/bin/bundle:23:in `load'
/usr/bin/bundle:23:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```
</details>

<details><summary>With CocoaPods in the <tt>Gemfile</tt></summary>

```sh
$ cat Gemfile 
# frozen_string_literal: true

source "https://rubygems.org"

gem "octokit"
gem "cocoapods"
gem "rake"
```

```sh
$ bundle install
Using rake 13.0.1
Using CFPropertyList 3.0.2
Using concurrent-ruby 1.1.6
Using i18n 0.9.5
Using minitest 5.14.1
Using thread_safe 0.3.6
Using tzinfo 1.2.7
Using activesupport 4.2.11.3
Using public_suffix 4.0.5
Using addressable 2.7.0
Using httpclient 2.8.3
Using json 2.3.1
Using algoliasearch 1.27.3
Using atomos 0.1.3
Using bundler 2.1.4
Using claide 1.0.3
Using fuzzy_match 2.0.4
Using nap 1.1.0
Using netrc 0.11.0
Using ffi 1.13.1
Using ethon 0.12.0
Using typhoeus 1.4.0
Using cocoapods-core 1.9.3
Using cocoapods-deintegrate 1.0.4
Using cocoapods-downloader 1.4.0
Using cocoapods-plugins 1.0.0
Using cocoapods-search 1.0.0
Using cocoapods-stats 1.1.0
Using cocoapods-trunk 1.5.0
Using cocoapods-try 1.2.0
Using colored2 3.1.2
Using escape 0.0.4
Using fourflusher 2.3.1
Using gh_inspector 1.1.3
Using molinillo 0.6.6
Using ruby-macho 1.4.0
Using nanaimo 0.3.0
Using xcodeproj 1.17.1
Using cocoapods 1.9.3
Using multipart-post 2.1.1
Using faraday 1.0.1
Using sawyer 0.8.2
Using octokit 4.18.0
Bundle complete! 3 Gemfile dependencies, 43 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

```sh
$ be rake test  
pod --version
1.9.3
bundle exec pod --version
1.9.3
```

</details>

Reply:
Cool, we should remove those from SwiftGen too then 👍